### PR TITLE
Bug31157

### PR DESCRIPTION
**Pull request by cohosh**
Implements https://trac.torproject.org/projects/tor/ticket/31157

### CODE REVIEW

File: broker/snowflake-broker_test.go [hunk 1]
```diff
@@ -408,7 +408,6 @@ func TestMetrics(t *testing.T) {
 			w := httptest.NewRecorder()
 			data := bytes.NewReader([]byte("{\"Sid\":\"ymbcCMto7KHNGYlp\",\"Version\":\"1.0\"}"))
 			r, err := http.NewRequest("POST", "snowflake.broker/proxy", data)
-			r.Header.Set("X-Session-ID", "test")
```
[the remark this answers is not in the document]
Comment (cohosh) [reply broker/snowflake-broker_test.go]: This isn't actually related to this commit, but we removed the use of `X-Session-ID` headers previously. See: https://gitweb.torproject.org/pluggable-transports/snowflake.git/commit/?id=c4ae64905b69512e50587c1ee749cddfc0937a4c

I caught that this was still there and figured I could remove it while I was modifying the file.

File: common/messages/proxy.go [hunk 2]
```diff
@@ -6,16 +6,18 @@ package messages
 import (
 	"encoding/json"
 	"fmt"
+	"strings"
 )
 
-const version = "1.0"
+const version = "1.1"
```
[the remark this answers is not in the document]
Comment (cohosh) [reply common/messages/proxy.go]: Yeah I was thinking of using major versions for changes that are not backwards compatible with proxies running a lower version.

File: broker/metrics.go [hunk 3]
```diff
@@ -19,6 +19,24 @@ We export metrics in the following format:
         A count of the total number of unique IP addresses of snowflake
         proxies that have polled.
 
+    "snowflake-ips-standalone" NUM NL
```
[the remark this answers is not in the document]
Comment (cohosh) [reply broker/metrics.go]: Agreed, I put it all in doc/broker-spec.txt and linked to it from metrics.go

File: broker/snowflake-heap.go [hunk 4]
```diff
@@ -10,6 +10,7 @@ over the offer and answer channels.
 */
 type Snowflake struct {
 	id            string
+	ptype         string
```
[the remark this answers is not in the document]
Comment (cohosh) [reply broker/snowflake-heap.go]: Yep, fixed :)

File: common/messages/proxy.go [hunk 5]
```diff
@@ -6,16 +6,18 @@ package messages
...
-/* Version 1.0 specification:
+/* Version 1.1 specification:
 
 == ProxyPollRequest ==
 {
-  Sid: [generated session id of proxy]
-  Version: 1.0
+  Sid: [generated session id of proxy],
+  Version: 1.1,
+  Type: [badge|webext|standalone]
```
[the remark this answers is not in the document]
Comment (cohosh) [reply common/messages/proxy.go]: Fixed.

File: common/messages/proxy.go [hunk 6]
```diff
@@ -73,34 +75,38 @@ HTTP 400 BadRequest
...
+func EncodePollRequest(sid string, ptype string) ([]byte, error) {
 	return json.Marshal(ProxyPollRequest{
 		Sid:     sid,
 		Version: version,
+		Type:    ptype,
 	})
 }
 
 // Decodes a poll message from a snowflake proxy and returns the
 // sid of the proxy on success and an error if it failed
```
[the remark this answers is not in the document]
Comment (cohosh) [reply common/messages/proxy.go]: Added.

**Comment by cohosh** [comment]
Okay I pushed an update, thanks!

File: broker/snowflake-broker_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (NullHypothesis) [reply broker/snowflake-broker_test.go]: Gotcha. I would suggest to move unrelated changes to separate commits in the future, to make the commit history easier to follow.

**Pull request closed without merging**